octopoes/tools/rename-origin-method.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+
+import json
+import logging
+
+import click
+from xtdb_client import XTDBClient
+
+logger = logging.getLogger(__name__)
+
+
+@click.group(
+    context_settings={
+        "help_option_names": ["-h", "--help"],
+        "max_content_width": 120,
+        "show_default": True,
+    }
+)
+@click.option("-n", "--node", default="0", help="XTDB node")
+@click.option(
+    "-u",
+    "--url",
+    default="http://localhost:3000",
+    help="XTDB server base url",
+)
+@click.option(
+    "-t",
+    "--timeout",
+    type=int,
+    default=5000,
+    help="XTDB request timeout (in ms)",
+)
+@click.option("-v", "--verbosity", count=True, help="Increase the verbosity level")
+@click.pass_context
+def cli(ctx: click.Context, url: str, node: str, timeout: int, verbosity: int):
+    verbosities = [logging.WARN, logging.INFO, logging.DEBUG]
+    try:
+        if verbosity:
+            logging.basicConfig(level=verbosities[verbosity - 1])
+    except IndexError:
+        raise click.UsageError("Invalid verbosity level (use -v, -vv, or -vvv)")
+
+    client = XTDBClient(url, node, timeout)
+    logger.info("Instantiated XTDB client with endpoint %s for node %s", url, node)
+
+    ctx.ensure_object(dict)
+    ctx.obj["client"] = client
+
+
+@cli.command(help="List observation origins based on method")
+@click.argument("method", required=False)
+@click.pass_context
+def list(ctx: click.Context, method: str):  # noqa: A001
+    if method:
+        query = f'{{:query {{:find [(pull ?var [*])] :where [[?var :type "Origin"][?var :origin_type "observation"][?var :method "{method}"]]}}}}'  # noqa: E501
+    else:
+        query = '{:query {:find [(pull ?var [*])] :where [[?var :type "Origin"][?var :origin_type "observation"][?var :method]]}}'  # noqa: E501
+    origins = ctx.obj["client"].query(query)
+    if not origins:
+        raise click.UsageError("No targets found")
+    if "error" in origins:
+        raise click.UsageError(origins["error"])
+    click.echo(json.dumps(origins))
+
+
+def search_replace_in_dict(ds, s: str, r: str):
+    for d in ds:
+        for k, v in d.items():
+            if s in v:
+                d[k] = v.replace(s, r)
+
+
+@cli.command(help="Rename an observation origin method")
+@click.option("--armed", is_flag=True)
+@click.argument("method")
+@click.argument("renamed")
```

Comment:
I think  "new_name" or something similar would be a better to understand name for this argument.

I had to read the code to understand what was meant with `armed`. CLI tools more often have a `--dry-run` / `-n` option. I think it would be better to also do that here because people are more familiar with that.

Reply:
It is not my aim to make this tool "user" friendly precisely because it should be used with care on special occasion by someone who has at least an idea of what he is doing... so if it incentivized you to read the code, it is probably a good thing, at least in my opinion

Reply:
The usual naming convention for mv's and similar tools is: source / destination. I think this would also fit this tool.
I agree with the Dry-run flag, however that unusual to be on by default, is that what we want, as the move is 'destructive' in the sense that we create new transactions  and thus not easily undone.